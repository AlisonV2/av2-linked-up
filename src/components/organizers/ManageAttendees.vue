<template>
  <div class="row">
    <div class="col-12" v-for="event in events" :key="event.id">
      <h2 class="title">{{ event.name }}</h2>
      <p>{{ event.attendees.length }}</p>
    </div>
  </div>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports ManageAttendees
 * @type {Component}
 * @vue-data {array} events
 * @vue-event created - Dispatch store action
 */
export default {
  name: 'ManageAttendees',
  data() {
    return {
      events: [],
      attendees: null,
    };
  },
  created() {
    this.$store
      .dispatch('getOrgaEvents')
      .then(() => {
        this.events = this.$store.getters.getOrgaEvents;
      })
      .catch((err) => {
        Sentry.captureException(err);
      });
  },
};
</script>
