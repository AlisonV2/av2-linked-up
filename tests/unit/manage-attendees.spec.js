import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import ManageAttendees from '@/components/organizers/ManageAttendees';

const actions = {
  getOrgaEvents: jest.fn(),
};

const getters = {
  getOrgaEvents: jest.fn(),
};
const store = createStore({
  actions,
  getters,
});

const config = {
  data() {
    return { 
        events: [], 
        attendees: null 
    };
  },
  global: {
    plugins: [store],
    components: {
      ManageAttendees,
    },
  },
};

/**
 * @module ManageAttendeesTest
 */
describe('ManageAttendees.vue', () => {
  it('ManageAttendees Snapshot', () => {
    const component = shallowMount(ManageAttendees, config);
    expect(component.element).toMatchSnapshot();
  });

  it('getAllArtists action is called on created', () => {
    const component = shallowMount(ManageAttendees, config);
    ManageAttendees.created.call(component.vm);
    expect(actions.getOrgaEvents).toHaveBeenCalled();
  });

  it('getAllArtists getters is called on created', () => {
    const component = shallowMount(ManageAttendees, config);
    ManageAttendees.created.call(component.vm);
    expect(getters.getOrgaEvents).toHaveBeenCalled();
  });
});
