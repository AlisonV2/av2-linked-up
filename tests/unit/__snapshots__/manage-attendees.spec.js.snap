// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageAttendees.vue ManageAttendees Snapshot 1`] = `
<div
  class="row"
>
  
  
</div>
`;
